app/src/main/java/com/mandysoftware/livrenouveau/DefinitionActivity.kt

```diff
@@ -0,0 +1,183 @@
+        val di = dictionary
+
+        // Get a reference to the action bar
+        supportActionBar?.apply {
+            // Enable the "up" button on the action bar
+            setDisplayHomeAsUpEnabled(true)
+
+        }
+
+
+
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        // Handle "up" button clicks
+        if (item.itemId == android.R.id.home) {
+            val intent = Intent(this@DefinitionActivity, SearchActivity::class.java)
+            intent.putExtra("word", word)
+            setResult(Activity.RESULT_OK, intent)
+            finish()
+
+            return true
+        }
+        return super.onOptionsItemSelected(item)
+    }
+
+    override fun onDictionaryError(command: CompositeDictionary.Command?, err: Exception) {
+        TODO("Not yet implemented")
+    }
+
+    override fun onDictionaryDownloadProgress(
+        dictionaryInfo: CompositeDictionary.DictionaryInfo,
+        bytesDownloaded: Int,
+        bytesTotal: Int
+    ) {
+        TODO("Not yet implemented")
+    }
+
+    override fun onDictionaryDownloadComplete(
+        dictionaryInfo: CompositeDictionary.DictionaryInfo,
+        success: Boolean,
+        err: Exception?
+    ) {
+        TODO("Not yet implemented")
+    }
+
+    override fun onDictionaryLoad(
+        dictionaryInfo: CompositeDictionary.DictionaryInfo,
+        success: Boolean,
+        err: Exception?
+    ) {
+        if(!success){
+            Toast.makeText(this, "Error loading dictionary.", Toast.LENGTH_SHORT).show()
+        } //maybe add else to lookup word
+    }
+
+    override fun onDictionaryDelete(
+        dictionaryInfo: CompositeDictionary.DictionaryInfo,
+        success: Boolean,
+        err: Exception?
+    ) {
+        TODO("Not yet implemented")
+    }
+
+    override fun onAllDictionariesReady(success: Boolean, err: Exception?) {
+        if(!success){
+            Toast.makeText(this, "Error loading ALL dictionary.", Toast.LENGTH_SHORT).show()
+        }
+
+    }
+
+    override fun onIndexDownloadProgress(bytesDownloaded: Int, bytesTotal: Int) {
+        //do nothing
+    }
+
+    override fun onIndexDownloadComplete(success: Boolean, err: Exception?) {
+        //do nothing
+    }
+
+    override fun onHeadFilesComplete(
+        successCount: Int,
+        totalCount: Int,
+        success: Boolean,
+        err: Exception?
+    ) {
+        TODO("Not yet implemented") //doesn't apply
+    }
+
+    override fun onLocalFilesChecked() {
+        //do nothing
+    }
+
+    override fun onDictionaryLookup(
+        results: MutableList<CompositeDictionary.LookupResult>,
+        successful: Int,
+        total: Int
+    ) {
+
+
+        for (result in results) {
+            if (result.isFound) {
+               // XWikiParser to start showing our wikitext as pretty outputs rather than not.
+                runOnUiThread {
+
+                    setTitle(result.word) //if user clicks on title take back to SearchActvity
```

Comment:
Clicking on the title currently doesn't do anything. Only clicking on the "left arrow" next to the title does.

Don't fix this in this PR, but for the future, we'll need to implement clicking on the title as well. Emma's Notebook does this title-clicking correctly, I'll have to look at it and tell you what I did there that worked.

Reply:
I looked into this, and I have a solution to make the title clickable. I have a set of changes that will do it, but I won't make my changes until we're done with your PR here. I'll push up a PR for you to review once we merge this PR.